### DISCUSSION TITLE
RAC Docs hard to interpret, a little frustrating

### DISCUSSION THREAD

**Discussion by binaryartifex** [adobe/react-spectrum#4905] [Q&A]
Im all in on RAC, however whenever there is a modicum of advancement needed, the docs seem to treat it with lip service at the end of the document and i think im honestly left more confused than had i not been exposed to the advanced use cases at all, im left with more questions than answers. Case in point, 'advanced customization' chapter at the end of the RAC menu page. i was trying to figure out how to completely separate the trigger so i could have an api like the following...

```typescript
      <Menu.Root
        aria-label="Account menu"
        disabledKeys={["organisation"]}
        onAction={(key) => console.log(`${key} fired`)}
      >
        <Menu.Trigger>
          <IconButton aria-label="Toggle dialog" icon={Icon.Article} />
        </Menu.Trigger>
        <Menu.List items={complexList}>
          {(section) => (
            <Menu.ListSection
              items={section.children}
              title={section.name}
            >
              {(item) => (
                <Menu.ListItem key={item.id} textValue={item.name}>
                  <Menu.ListOption
                    description={item.description}
                    icon={item.icon}
                    name={item.name}
                  />
                </Menu.ListItem>
              )}
            </Menu.ListSection>
          )}
        </Menu.List>
      </Menu.Root>
```

now truth be told i managed to pull off the above with react-aria hooks, however the implementation doesn't pass my 'hit by a truck' principle - ie. if i got hit by a truck tomorrow, could anyone pick up where i left off or would a rewrite be needed purely because it'd take too long to solve that rubix cube?

Now the bottom of the menu page shows something very interesting,  a Provider component thats only ever mentioned in passing but looks like it could provide a perfect intermediary between raw hooks and vanilla RAC. but it'd be great if a little more attention was given with typescript support and full use cases and have it explained to me like im 5. at the moment im no closer to achieving the RAC equivalent of the above API

**Comment by devongovett** [comment]
Yeah the "Advanced customization" section is currently a very early work in progress and will be getting a re-think soon.

**Reply by devongovett** [reply]
FYI this has now been updated across all components. Let me know if it is helpful.
